Feature expedition

nowickiark opened the pull request.

Commit:
Added expedition and truck feature.

Commit:
Added expedition and truck feature.
-fiexed views and editing of entities

CKochanski commented on mvnw:
```diff
@@ -0,0 +1,286 @@
+#!/bin/sh
```
Nie chcemy tego pliku

CKochanski commented on mvnw.cmd:
```diff
@@ -0,0 +1,161 @@
+@REM ----------------------------------------------------------------------------
```
Tego też nie

CKochanski commented on src/main/java/com/sda/groupa/shippingcostcalculator/expedition/controller/ExpeditionController.java:
```diff
@@ -0,0 +1,79 @@
+@Controller
+public class ExpeditionController {
+
+    private ExpeditionService expeditionService;
```
Pola niech będą immutable (final)

CKochanski commented on src/main/java/com/sda/groupa/shippingcostcalculator/expedition/controller/ExpeditionController.java:
```diff
@@ -0,0 +1,79 @@
+        this.truckService = truckService;
+    }
+
+    @GetMapping("/expedition/Add")
```
`/expedition/add`

nowickiark replied:
fixed

CKochanski commented on src/main/java/com/sda/groupa/shippingcostcalculator/expedition/controller/ExpeditionController.java:
```diff
@@ -0,0 +1,79 @@
+    public ModelAndView getExpeditionform(@PathVariable Long id){
+        ModelAndView modelAndView = new ModelAndView("expedition");
+
+        Expedition expedition = expeditionService.getExpeditionById(id).get();
```
Mamy pewność, że dostaniemy id, które istnieje?

nowickiark replied:
added .orElseThrowException

CKochanski commented on src/main/java/com/sda/groupa/shippingcostcalculator/expedition/controller/ExpeditionController.java:
```diff
@@ -0,0 +1,79 @@
+            return modelAndView;
+    }
+
+    @PostMapping("/expedition/Add")
```
`/expedition/add`

CKochanski commented on src/main/java/com/sda/groupa/shippingcostcalculator/expedition/controller/ExpeditionController.java:
```diff
@@ -0,0 +1,79 @@
+
+        Expedition expedition1 = expeditionService.getExpeditionById(expedition.getId()).get();
+
+        System.out.println(expedition.getId());
```
No printlns in production code :)

CKochanski commented on src/main/java/com/sda/groupa/shippingcostcalculator/expedition/model/Expedition.java:
```diff
@@ -0,0 +1,165 @@
+    private LocalDate startDay;
+    private LocalDate endDay;
+
+    private BigDecimal CashBeginingZl;
```
Pola z małej litery

CKochanski commented on src/main/java/com/sda/groupa/shippingcostcalculator/expedition/model/Expedition.java:
```diff
@@ -0,0 +1,165 @@
+        this.endOdometerReading = endOdometerReading;
+        this.startDay = startDay;
+        this.endDay = endDay;
+        CashBeginingZl = cashBeginingZl;
```
Danger!

CKochanski commented on src/main/java/com/sda/groupa/shippingcostcalculator/expedition/service/ExpeditionService.java:
```diff
@@ -0,0 +1,35 @@
+@Service
+public class ExpeditionService {
+
+   private ExpeditionRepository expeditionRepository;
```
Immutable (final)

CKochanski commented on src/main/java/com/sda/groupa/shippingcostcalculator/expedition/service/ExpeditionService.java:
```diff
@@ -0,0 +1,35 @@
+   private ExpeditionRepository expeditionRepository;
+
+   @Autowired
+    public ExpeditionService(ExpeditionRepository expeditionRepository){
```
Od springa 4.3 nie musimy dawać na konstruktorze `@Autowired`

CKochanski commented on src/main/java/com/sda/groupa/shippingcostcalculator/truck/controller/TruckController.java:
```diff
@@ -0,0 +1,67 @@
+@Controller
+public class TruckController {
+
+    private TruckService truckService;
```
Immutable :)

CKochanski commented on src/main/java/com/sda/groupa/shippingcostcalculator/truck/controller/TruckController.java:
```diff
@@ -0,0 +1,67 @@
+        return modelAndView;
+    }
+
+    @GetMapping("/truck/Add")
```
`/truck/add`

CKochanski commented on src/main/java/com/sda/groupa/shippingcostcalculator/truck/controller/TruckController.java:
```diff
@@ -0,0 +1,67 @@
+    }
+
+
+    @GetMapping("/truck/Add/{id}")
```
`/truck/add/{id}`

CKochanski commented on src/main/java/com/sda/groupa/shippingcostcalculator/truck/controller/TruckController.java:
```diff
@@ -0,0 +1,67 @@
+        return modelAndView;
+    }
+
+    @PostMapping("/truck/Add")
```
`/truck/add`

CKochanski commented on src/main/java/com/sda/groupa/shippingcostcalculator/truck/service/TruckService.java:
```diff
@@ -0,0 +1,41 @@
+@Service
+public class TruckService {
+
+    public TruckRepository truckRepository;
```
Immutable

nowickiark replied:
fixed

CKochanski commented on src/main/webapp/WEB-INF/views/expedition.jsp:
```diff
@@ -0,0 +1,52 @@
+            <td><form:input path="startDay"/></td>
+        </tr>
+        <tr>
+            <td><form:label path="CashBeginingZl">Amount of cash in zloty</form:label></td>
```
Uważaj na duże litery przy refactorze

nowickiark replied:
fixed

CKochanski commented on src/main/webapp/WEB-INF/views/expeditions.jsp:
```diff
@@ -0,0 +1,37 @@
+                <td>${expedition.startingPlace}</td>
+                <td>${expedition.startOdometerReading}</td>
+                <td>${expedition.startDay}</td>
+                <td><a href="/expedition/Add/${expedition.id}">Edit</a></td>
```
Po zmianie endpointa trzeba zmienić tego hrefa

nowickiark replied:
fixed

CKochanski commented on src/main/webapp/WEB-INF/views/expeditions.jsp:
```diff
@@ -0,0 +1,37 @@
+
+    </table>
+
+    <a href="/expedition/Add">Add expedition</a>
```
Tu też :)

nowickiark replied:
fixed

CKochanski commented on src/main/webapp/WEB-INF/views/truckList.jsp:
```diff
@@ -0,0 +1,37 @@
+                <td>${truck.brand}</td>
+                <td>${truck.model}</td>
+                <td>${truck.horsePower}</td>
+                <td><a href="/truck/Add/${truck.id}">Truck Details</a></td>
```
I tu :)

nowickiark replied:
fixed

Commit:
Fixes for review.
Changed in Add to add.
Added veryfication for optional.
Changed repositories and service as final.

Commit:
Fixes for review.
Changed in Add to add.
Added veryfication for optional.
Changed repositories and service as final.

Commit:
Fixed date adding

nowickiark committed:
Merge branch 'master' into feature_Expedition
